### DISCUSSION TITLE
Obsidian integration?

### DISCUSSION THREAD

**Discussion by abdurraheemali** [idyll-lang/idyll#782] [Ideas]
It would be nice to be able to have interactive idyll components in Obsidian. Is this technically feasible?

**Comment by mathisonian** [comment]
Hi @abdurraheemali,

I'm not too familiar with Obsidian. Do you know what kind of API they expose? If its possible to render custom JavaScript / React components there it may be feasible to do.

**Reply by abdurraheemali** [reply]
https://obsidian.md/plugins Has a good overview of the 531 community plugins. They're all on Github so it's possible to check the source directly. https://forum.obsidian.md/ has some previous discussions. Templater, one of the most popular ones, does let you execute arbitrary javascript. 

The pragmatic argument for building an Obsidian plugin is audience-building. For instance, the Excalidraw plugin is a traffic generating funnel to the Excalidraw+ service. However, since idyll isn't (to my knowledge) monetized, that kinda sounds like "working for exposure", so I wouldn't prioritize it over core features + stability improvements that you're wanting to implement. Or even if you _were_ optimizing for popularity, maybe you'd have better options?

I personally would be far more likely to use Idyll if it was available as a plugin, but I'm just one person, maybe I'm just being a source of randomization here. So really a selfish feature request :)
